### PR TITLE
OHRM5X-1003: Employee Timesheet get All

### PR DESCRIPTION
**Pull request by SandeepaOHRM**
This PR contains:

- Backend API for employee timesheet get all
- Intergration testings for the API

### CODE REVIEW

File: installer/dbscript/dbscript-2.sql [hunk 1]
```diff
@@ -4258,7 +4258,10 @@ VALUES ('apiv2_time_time_sheet_config', 'API-v2 Time - Time Sheet Start Day Conf
        ('apiv2_time_timesheet_comment', 'API-v2 Time - Timesheet comment', 1, 1, 1, 0),
        ('apiv2_time_config_time_format', 'API-v2 Time - Time Config Format', 1, 0, 0, 0),
        ('apiv2_time_validation_project_activity_name', 'API-v2 Time - Project Activity Name Validation', 1, 0, 0, 0),
-       ('apiv2_time_project_activity_unique_validation', 'API-v2 Time - Project Activity Unique Validation', 1, 0, 0, 0);
+       ('apiv2_time_project_activity_unique_validation', 'API-v2 Time - Project Activity Unique Validation', 1, 0, 0, 0),
+       ('apiv2_time_employee_timesheets', 'API-v2 Time - Employee Timesheets', 1, 0, 0, 0);
```
Comment (RajithaKumara) [inline installer/dbscript/dbscript-2.sql:4262]:
```suggestion
       ('apiv2_time_employee_timesheets', 'API-v2 Time - Employee Timesheets', 1, 1, 1, 0);
```
check `apiv2_time_my_timesheet`

File: installer/dbscript/dbscript-2.sql [hunk 2]
```diff
@@ -4315,7 +4320,9 @@ VALUES (1, 0, 1, 0, 0, @apiv2_time_time_sheet_config_data_group_id, @admin_role_
        (1, 0, 0, 0, 0, @apiv2_time_config_time_format_data_group_id, @admin_role_id),
        (1, 0, 0, 0, 0, @apiv2_time_config_time_format_data_group_id, @ess_role_id),
        (1, 0, 0, 0, 0, @apiv2_time_validation_project_activity_name_data_group_id, @admin_role_id),
-       (1, 0, 0, 0, 0, @apiv2_time_validation_project_activity_name_data_group_id, @project_admin_role_id);
+       (1, 0, 0, 0, 0, @apiv2_time_validation_project_activity_name_data_group_id, @project_admin_role_id),
+       (1, 0, 0, 0, 0, @apiv2_time_employee_timesheets_data_group_id, @admin_role_id),
+       (1, 0, 0, 0, 0, @apiv2_time_employee_timesheets_data_group_id, @supervisor_role_id);
```
Comment (RajithaKumara) [inline installer/dbscript/dbscript-2.sql:4325]:
```suggestion
       (1, 1, 1, 0, 0, @apiv2_time_employee_timesheets_data_group_id, @admin_role_id),
       (1, 1, 1, 0, 0, @apiv2_time_employee_timesheets_data_group_id, @supervisor_role_id);
```
@VageeshOrangeHRM check this

File: symfony/plugins/orangehrmTimePlugin/Api/EmployeeTimesheetAPI.php [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+                $this->getDate()
+            );
+            $timesheetParamHolder->setFromDate(new DateTime($fromDate));
+            $timesheetParamHolder->setToDate(new DateTime($toDate));
+        } elseif (is_null($this->getFromDateParam()) && is_null($this->getDate())) {
+            list($fromDate, $toDate) = $this->getTimesheetService()->extractStartDateAndEndDateFromDate(
+                $this->getDateTimeHelper()->getNow()
+            );
+            $timesheetParamHolder->setFromDate(new DateTime($fromDate));
+            $timesheetParamHolder->setToDate(new DateTime($toDate));
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmTimePlugin/Api/EmployeeTimesheetAPI.php:184]: will reconsider this

File: symfony/plugins/orangehrmTimePlugin/test/Api/TimesheetAPITest.php [hunk 4]
```diff
@@ -24,6 +24,7 @@
 use OrangeHRM\Tests\Util\EndpointIntegrationTestCase;
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmTimePlugin/test/Api/TimesheetAPITest.php:24]: Rename this class to `EmployeeTimesheetAPITest`

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: installer/dbscript/dbscript-2.sql (hunk 1 above)
Comment (SandeepaOHRM) [reply installer/dbscript/dbscript-2.sql:4262]: done

File: installer/dbscript/dbscript-2.sql (hunk 2 above)
Comment (SandeepaOHRM) [reply installer/dbscript/dbscript-2.sql:4325]: done

File: symfony/plugins/orangehrmTimePlugin/Api/EmployeeTimesheetAPI.php (hunk 3 above)
Comment (SandeepaOHRM) [reply symfony/plugins/orangehrmTimePlugin/Api/EmployeeTimesheetAPI.php:184]: sure

File: symfony/plugins/orangehrmTimePlugin/test/Api/TimesheetAPITest.php (hunk 4 above)
Comment (SandeepaOHRM) [reply symfony/plugins/orangehrmTimePlugin/test/Api/TimesheetAPITest.php:24]: done

File: symfony/plugins/orangehrmTimePlugin/Api/EmployeeTimesheetAPI.php [hunk 5]
```diff
@@ -0,0 +1,387 @@
...
+    public const PARAMETER_COMMENT = 'comment';
+
+    public const FILTER_FROM_DATE = 'fromDate';
+    public const FILTER_TO_DATE = 'toDate';
+    public const FILTER_DATE = 'date';
+
+    /**
+     * @inheritDoc
+     * @throws BadRequestException
+     * @throws Exception
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmTimePlugin/Api/EmployeeTimesheetAPI.php:76]:
```suggestion
```

File: symfony/plugins/orangehrmTimePlugin/Api/EmployeeTimesheetAPI.php [hunk 6]
```diff
@@ -0,0 +1,387 @@
...
+     * @inheritDoc
+     */
+    public function getValidationRuleForCreate(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            $this->getValidationDecorator()->requiredParamRule(
+                new ParamRule(
+                    self::PARAMETER_DATE,
+                    new Rule(Rules::API_DATE),
+                    new Rule(MyTimesheetDateRule::class),
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmTimePlugin/Api/EmployeeTimesheetAPI.php:262]:
```suggestion
                    new Rule(TimesheetDateRule::class),
```

File: symfony/plugins/orangehrmTimePlugin/Api/EmployeeTimesheetAPI.php [hunk 7]
```diff
@@ -0,0 +1,387 @@
...
+                new ParamRule(
+                    self::PARAMETER_ACTION,
+                    new Rule(Rules::STRING_TYPE),
+                    new Rule(MyTimesheetActionRule::class),
+                )
+            ),
+            $this->getValidationDecorator()->notRequiredParamRule(
+                new ParamRule(
+                    self::PARAMETER_COMMENT,
+                    new Rule(Rules::STRING_TYPE),
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmTimePlugin/Api/EmployeeTimesheetAPI.php:365]: seems length validation missing

**Review by RajithaKumara — APPROVED** [review]

File: symfony/plugins/orangehrmTimePlugin/Api/EmployeeTimesheetAPI.php (hunk 5 above)
Comment (SandeepaOHRM) [reply symfony/plugins/orangehrmTimePlugin/Api/EmployeeTimesheetAPI.php:76]: Done

File: symfony/plugins/orangehrmTimePlugin/Api/EmployeeTimesheetAPI.php (hunk 6 above)
Comment (SandeepaOHRM) [reply symfony/plugins/orangehrmTimePlugin/Api/EmployeeTimesheetAPI.php:262]: Done

File: symfony/plugins/orangehrmTimePlugin/Api/EmployeeTimesheetAPI.php (hunk 7 above)
Comment (SandeepaOHRM) [reply symfony/plugins/orangehrmTimePlugin/Api/EmployeeTimesheetAPI.php:365]: In database there is no defined length

**Pull request merged**